File: src/main/java/net/golem/raknet/Identifier.java
package net.golem.raknet;

import java.util.ArrayList;

public abstract class Identifier {

	private static String HEADER = "MCPE";
	private static char SEPARATOR = ';';

	public abstract ArrayList<Object> getValues();

	public String build() {
		StringBuilder builder = new StringBuilder();
		builder.append(HEADER).append(SEPARATOR);
		getValues().forEach(value -> builder.append(value).append(SEPARATOR));
		return builder.toString();
	}
}

File: src/main/kotlin/raknet/Identifier.kt
package raknet

class Identifier(private vararg val values: Any) {

    override fun toString(): String {
        val builder = StringBuilder()
        builder.append(HEADER)
        builder.append(SEPARATOR)
        for (value in values) {
            builder.append(value)
            builder.append(SEPARATOR)
        }
        return builder.toString()
    }

    companion object {
        const val HEADER: String = "MCPE"
        const val SEPARATOR: Char = ';'
    }

}